solutions/USACO jan17-not-last.mdx

```diff
@@ -36,230 +36,44 @@ Henrietta 4
-	return res; }
-tcTU> str ts(pair<T,U> p);
-tcT> str ts(T v) { // containers with begin(), end()
-	#ifdef LOCAL
-		bool fst = 1; str res = "{";
-		for (const auto& x: v) {
-			if (!fst) res += ", ";
-			fst = 0; res += ts(x);
-		}
-		res += "}"; return res;
-	#else
-		bool fst = 1; str res = "";
-		for (const auto& x: v) {
-			if (!fst) res += " ";
-			fst = 0; res += ts(x);
-		}
-		return res;
-
-	#endif
-}
-tcTU> str ts(pair<T,U> p) {
-	#ifdef LOCAL
-		return "("+ts(p.f)+", "+ts(p.s)+")"; 
-	#else
-		return ts(p.f)+" "+ts(p.s);
-	#endif
-}
-
-// OUTPUT
-tcT> void pr(T x) { cout << ts(x); }
-tcTUU> void pr(const T& t, const U&... u) { 
-	pr(t); pr(u...); }
-void ps() { pr("\n"); } // print w/ spaces
-tcTUU> void ps(const T& t, const U&... u) { 
-	pr(t); if (sizeof...(u)) pr(" "); ps(u...); }
-
-// DEBUG
-void DBG() { cerr << "]" << endl; }
-tcTUU> void DBG(const T& t, const U&... u) {
-	cerr << ts(t); if (sizeof...(u)) cerr << ", ";
-	DBG(u...); }
-#ifdef LOCAL // compile with -DLOCAL, chk -> fake assert
-	#define dbg(...) cerr << "Line(" << __LINE__ << ") -> [" << #__VA_ARGS__ << "]: [", DBG(__VA_ARGS__)
-	#define chk(...) if (!(__VA_ARGS__)) cerr << "Line(" << __LINE__ << ") -> function(" \
-		 << __FUNCTION__  << ") -> CHK FAILED: (" << #__VA_ARGS__ << ")" << "\n", exit(0);
-#else
-	#define dbg(...) 0
-	#define chk(...) 0
-#endif
+using namespace std;
 
-void setPrec() { cout << fixed << setprecision(15); }
-void unsyncIO() { cin.tie(0)->sync_with_stdio(0); }
-// FILE I/O
-void setIn(str s) { freopen(s.c_str(),"r",stdin); }
-void setOut(str s) { freopen(s.c_str(),"w",stdout); }
-void setIO(str s = "") {
-	unsyncIO(); setPrec();
-	// cin.exceptions(cin.failbit); 
-	// throws exception when do smth illegal
-	// ex. try to read letter into int
-	if (sz(s)) setIn(s+".in"), setOut(s+".out"); // for USACO
-}
+constexpr int COW_NUM = 7;
 
+// detemplating courtesy of kevin sheng
 int main() {
-	setIO("notlast");
-	int N; re(N);
-	map<str,int> tot;
-	F0R(i,N) {
-		str a; int b; re(a,b);
-		tot[a] += b;
+	ifstream read("notlast.in");
+	int N;
+	read >> N;
+
+	map<string, int> raw;
+	for (int i = 0; i < N; i++) {
+		string a;
+		int b;
+		read >> a >> b;
+		raw[a] += b;
+	}
+
+	vector<pair<int, string>> cows;
+	for (pair<string, int> t : raw) {
+		cows.push_back({t.second, t.first});
 	}
-	V<pair<int,str>> cows;
-	trav(t,tot) cows.pb({t.s,t.f});
-	sort(all(cows));
+	sort(cows.begin(), cows.end());
+
 	int ind = 0;
-	if (sz(cows) == 7) {
-		while (ind < sz(cows) && cows[ind].f == cows[0].f) ind ++;
+	// only move the pointer if all cows produced some milk- any unmentioned cows will have produced 0 milk
+	if (cows.size() == COW_NUM) {
+		while (ind < cows.size() && cows[ind].first == cows[0].first) {
+			ind++;
+		}
 	}
-	if (ind < sz(cows) && (ind+1 == sz(cows) || cows[ind].f != cows[ind+1].f)) {
-		ps(cows[ind].s);
-		exit(0);
+
+	if (ind < cows.size() && (ind + 1 == cows.size() || cows[ind].first != cows[ind + 1].first)) {
```

Comment:
Also too long, break it into two lines

Reply:
I mean, it isn't long enough for the website to wrap it, so I think this one is fine.

Reply:
?? Maybe you have a larger screen than me, but it wraps on my screen. It's longer than 80 characters, and lines should not exceed 80 characters as mentioned [here](https://usaco.guide/general/contributing?lang=cpp#code-conventions).

Reply:
Ah, I see. I'll break it up then.